kfdef/odh-core-experimental.yaml

```diff
@@ -0,0 +1,73 @@
+# The components listed in this KfDef are pulling the latest kustomize manifest and images directly from each components development repo
+# If you deploy odh-core using this KfDef there is no expectation for stability and components made be updated without any prior notice
+apiVersion: kfdef.apps.kubeflow.org/v1
+kind: KfDef
+metadata:
+  name: odh-core-experimental
+spec:
+  applications:
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: odh-common
+    name: odh-common
+  - kustomizeConfig:
+      repoRef:
+        name: manifests-dashboard  # Use the odh-dashboard repo as the source for this kustomize manifest
+        path: manifests
+    name: odh-dashboard
+  #TODO: Resolve issues with kustomize parsing the odh-controller manifest
+  # Deploy the KF notebook controller and ODH controller separately since they are normally bundled into the
+  #   same manifests deployment when synchronized to odh-manifests
+  #- kustomizeConfig:
+  #    overlays:
+  #      - openshift
+  #    repoRef:
+  #      name: manifests-notebook-controller
+  #      path: components/notebook-controller/overlays/openshift
+  #  name: kf-notebook-controller
+  #- kustomizeConfig:
+  #    repoRef:
+  #      name: manifests-notebook-controller
+  #      path: components/odh-notebook-controller
+  #  name: odh-notebook-controller
+
+  # Pull from odh-manifests since the notebook-controller manifest in the downstream kubeflow repo
+  #   still requires customizations for a successful deployment with the odh-operator
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: odh-notebook-controller
+    name: odh-notebook-controller
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: notebook-images
+    name: notebook-images
+  - kustomizeConfig:
+      repoRef:
+        name: manifests-model-mesh
+        path: manifests/opendatahub/overlays/odh-model-controller
```

Comment:
@LaVLaS I just did a clean installation again and my modelmesh-controller was missing.  Could you confirm with your deployment?  I'm reverting my modelmesh deployment to core for now.

Reply:
We're in the process of updating all the manifests to get them clean again -- in the 1.20 rush the repos ended up in a bad state. I'll drop a comment in here once it's ready / raise a new PR later if this gets merged in.

Reply:
@anishasthana To simplify the process, if you focus on getting the manifest in the MLOps code repos working with as an "experimental" release then the only change required to include is as an "odh-core" release should be copy/paste into odh-manifests and updating image tags from a `latest` to a static tag for a stable release

Reply:
@anishasthana @LaVLaS Before you merge this, I had to use use a separate `overlays` parameter to get it to deploy the modelmesh-controller instead of using the `path: manifests/opendatahub/overlays/odh-model-controller`:
```
    - kustomizeConfig:
        overlays:
          - odh-model-controller
        repoRef:
          name: manifests
          path: manifests/opendatahub
      name: model-mesh
```
Does this need to change in this kfdef?

Reply:
@anishasthana is https://github.com/opendatahub-io/modelmesh-serving/pull/60 the appropriate PR to test with this experimental kfdef?

Reply:
https://github.com/opendatahub-io/modelmesh-serving/pull/60 has not been tested yet, https://github.com/opendatahub-io/modelmesh-serving/pull/55 has been and passed the test. However, yesterday we re-looked at all the commits, and decided to change the approach to syncing a bit since we noticed the manifests between odh-manifests and model-mesh repos are not in sync either (ie. within upstream). So currently I am working on making sure the manifests are identical before merging https://github.com/opendatahub-io/modelmesh-serving/pull/55.

Reply:
@heyselbi @anishasthana any progress here?